File: bridge/lib/web/bridge_remote_config.js
let js_bridge_remoteConfig = {
    js_bridge_remoteConfig_isSupported: function () {
        return bridge.remoteConfig.isSupported;
    },

    js_bridge_remoteConfig_setDynamicParameters: function (parameters) {
        var jsParameters = JSON.parse(UTF8ToString(parameters));
        bridge.remoteConfig.setDynamicParameters(jsParameters);
    },

    js_bridge_remoteConfig_get: function (handler, onSuccess, onFailure) {
        bridge.remoteConfig.get()
            .then(data => {
                {{{ makeDynCall('viiii', 'handler') }}} (onSuccess, onFailure, 0, packToJson(data));
            })
            .catch(error => {
                {{{ makeDynCall('viiii', 'handler') }}} (onSuccess, onFailure, 1, packToJson(error));
            })
    },
}

mergeInto(LibraryManager.library, js_bridge_remoteConfig);

File: bridge/src/bridge_remote_config.cpp
#if defined(DM_PLATFORM_HTML5)
#include "bridge_remote_config.h"
#include "bridge.h"

int bridge::remoteConfig::isSupported(lua_State* L) {
    return getBoolean(L, js_bridge_remoteConfig_isSupported);
}

int bridge::remoteConfig::setDynamicParameters(lua_State* L) {
    return callWithJson(L, js_bridge_remoteConfig_setDynamicParameters);
}

int bridge::remoteConfig::get(lua_State* L) {
    return makeCallback(L, js_bridge_remoteConfig_get, true);
}

#endif